spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalog.java

```diff
@@ -52,32 +54,104 @@ public class GravitinoCatalog implements TableCatalog, SupportsNamespaces {
+    this.gravitinoAdaptor = GravitinoCatalogAdaptorFactory.createGravitinoAdaptor(provider);
+    this.sparkCatalog =
+        gravitinoAdaptor.createAndInitSparkCatalog(
+            name, options, gravitinoCatalogClient.properties());
+    this.propertiesConverter = gravitinoAdaptor.getPropertiesConverter();
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
   @Override
   public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException {
-    throw new NotSupportedException("Doesn't support listing table");
+    String gravitinoNamespace;
+    if (namespace.length == 0) {
+      gravitinoNamespace = getCatalogDefaultNamespace();
+    } else {
+      validateNamespace(namespace);
+      gravitinoNamespace = namespace[0];
+    }
+    try {
+      NameIdentifier[] identifiers =
+          gravitinoCatalogClient
+              .asTableCatalog()
+              .listTables(Namespace.of(metalakeName, catalogName, gravitinoNamespace));
+      return Arrays.stream(identifiers)
+          .map(
+              identifier ->
+                  Identifier.of(new String[] {getDatabase(identifier)}, identifier.name()))
+          .toArray(Identifier[]::new);
+    } catch (NoSuchSchemaException e) {
+      throw new NoSuchNamespaceException(namespace);
+    }
   }
 
   @Override
   public Table createTable(
       Identifier ident, Column[] columns, Transform[] partitions, Map<String, String> properties)
       throws TableAlreadyExistsException, NoSuchNamespaceException {
-    throw new NotSupportedException("Doesn't support creating table");
+    NameIdentifier gravitinoIdentifier =
+        NameIdentifier.of(metalakeName, catalogName, getDatabase(ident), ident.name());
+    com.datastrato.gravitino.rel.Column[] gravitinoColumns =
+        Arrays.stream(columns)
+            .map(column -> createGravitinoColumn(column))
+            .toArray(com.datastrato.gravitino.rel.Column[]::new);
+
+    Map<String, String> gravitinoProperties =
+        propertiesConverter.toGravitinoTableProperties(properties);
+    // Spark store comment in properties, we should retrieve it and pass to Gravitino explicitly.
+    String comment = gravitinoProperties.remove(ConnectorConstants.COMMENT);
+
+    try {
+      com.datastrato.gravitino.rel.Table table =
+          gravitinoCatalogClient
+              .asTableCatalog()
+              .createTable(gravitinoIdentifier, gravitinoColumns, comment, gravitinoProperties);
+      return gravitinoAdaptor.createSparkTable(ident, table, sparkCatalog, propertiesConverter);
+    } catch (NoSuchSchemaException e) {
+      throw new NoSuchNamespaceException(ident.namespace());
+    } catch (com.datastrato.gravitino.exceptions.TableAlreadyExistsException e) {
+      throw new TableAlreadyExistsException(ident);
+    }
   }
 
-  // Will create a catalog specific table by invoking createSparkTable()
+  // Will create a catalog specific table
```

Comment:
Is this comment valid?

Reply:
If you're specifying line 151, probably you need to add comment to that line.

Reply:
moved to 151